app/backend/prepdocslib/searchmanager.py

```diff
@@ -65,140 +67,190 @@ def __init__(
+                                ),
+                            )
+                        ]
+                    ),
+                    vector_search=VectorSearch(
+                        algorithms=[
+                            HnswAlgorithmConfiguration(
+                                name="hnsw_config",
+                                parameters=HnswParameters(metric="cosine"),
+                            )
+                        ],
+                        profiles=[
+                            VectorSearchProfile(
+                                name="embedding_config",
+                                algorithm_configuration_name="hnsw_config",
+                                vectorizer_name=(
+                                    f"{self.search_info.index_name}-vectorizer" if self.use_int_vectorization else None
+                                ),
                             ),
-                        ),
-                    ],
-                    vectorizers=vectorizers,
-                ),
-            )
-            if self.search_info.index_name not in [name async for name in search_index_client.list_index_names()]:
-                logger.info("Creating %s search index", self.search_info.index_name)
+                        ],
+                        vectorizers=vectorizers,
+                    ),
+                )
+
                 await search_index_client.create_index(index)
             else:
                 logger.info("Search index %s already exists", self.search_info.index_name)
-                index_definition = await search_index_client.get_index(self.search_info.index_name)
-                if not any(field.name == "storageUrl" for field in index_definition.fields):
+                existing_index = await search_index_client.get_index(self.search_info.index_name)
+                if not any(field.name == "storageUrl" for field in existing_index.fields):
                     logger.info("Adding storageUrl field to index %s", self.search_info.index_name)
-                    index_definition.fields.append(
+                    existing_index.fields.append(
                         SimpleField(
                             name="storageUrl",
                             type="Edm.String",
                             filterable=True,
                             facetable=False,
                         ),
                     )
-                    await search_index_client.create_or_update_index(index_definition)
+                    await search_index_client.create_or_update_index(existing_index)
+
+                if existing_index.vector_search is not None and (
+                    existing_index.vector_search.vectorizers is None
+                    or len(existing_index.vector_search.vectorizers) == 0
+                ):
+                    if self.embeddings is not None and isinstance(self.embeddings, AzureOpenAIEmbeddingService):
```

Comment:
This is new code, adding a vectorizer if one doesnt exist yet, to make it easy for folks to upgrade